pkg/services/ngalert/api/prometheus/api_prometheus.go

```diff
@@ -899,18 +905,20 @@ func PrepareRuleGroupStatuses(log log.Logger, store ListAlertRulesStore, opts Ru
 
 	receiverName := opts.Query.Get("receiver_name")
 	title := opts.Query.Get("search.rule_name")
+	searchDataSources := opts.Query["search.datasources"]
```

Comment:
Let's name it `datasource_uid` instead? Parameters with the `search.` prefix currently do partial case-insensitive matches, but here we're doing an exact match on datasource UID.